flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/SchedulingStrategyUtils.java

```diff
@@ -37,22 +38,28 @@ static List<ExecutionVertexID> sortExecutionVerticesInTopologicalOrder(
                 .collect(Collectors.toList());
     }
 
-    static List<SchedulingPipelinedRegion> sortPipelinedRegionsInTopologicalOrder(
+    /**
+     * Sort pipelined regions in topological order.
+     *
+     * @param topology represent the graph contains all pipelined regions.
+     * @param regions to be sorted in topological order.
+     * @return regions in topological order, the set must preserve this order.
+     */
+    static Set<SchedulingPipelinedRegion> sortPipelinedRegionsInTopologicalOrder(
```

Comment:
This method should explicitly returns a `LinkedHashSet`, indicating that it's ordered.

Reply:
good suggestion.